model/shared/src/main/scala/explore/model/Asterism.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats.Eq
+import cats.data.NonEmptyList
+import cats.syntax.all._
+import lucuma.core.model.Target
+import monocle._
+
+final case class Asterism(targets: NonEmptyList[TargetWithId]) {
```

Comment:
Would it make sense to internally use a `Map[Target.Id, Target]`? We should use a `Map` implementation that preserves insertion order, like `VectorMap` or  `TreeSeqMap`. However, I'm not sure there are "`NonEmpty`" versions of those out of the box.

The `targetOptional` method for building `Optional` wouldn't be needed then, it would just be the standard `index` optic.

Reply:
Right, there is a NonEmptyMap but I don't think it cares about order. Maybe a NonEmptySet to ensure ids are unique?

Reply:
I'm going to punt in this question and perhaps address it later. In the mean time I'm hidding the detail that we are using `NonEmptyList` thus changing it will be easier in the future